salvo/README.md

```diff
@@ -0,0 +1,73 @@
+# salvo
+
+This is a framework that abstracts executing multiple benchmarks of the Envoy Proxy using [NightHawk](https://github.com/envoyproxy/nighthawk).
+
+## Example Control Documents
+
+The control document defines the data needed to excute a benchmark.  At the moment the dockerized scavenging benchmark is the only one supported.  To run the benchmark, create a file with the following example contents:
+
+JSON Example:
+
+```json
+{
+  "remote": false,
+  "dockerizedBenchmark": true,
+  "images": {
+    "reuseNhImages": true,
+    "nighthawkBenchmarkImage": "envoyproxy/nighthawk-benchmark-dev:latest",
+    "nighthawkBinaryImage": "envoyproxy/nighthawk-dev:latest",
+    "envoyImage": "envoyproxy/envoy-dev:f61b096f6a2dd3a9c74b9a9369a6ea398dbe1f0f"
+  },
+  "environment": {
+    "v4only": true,
+    "envoyPath": "envoy",
+    "outputDir": "/home/ubuntu/nighthawk_output",
+    "testDir": "/home/ubuntu/nighthawk_tests"
+  }
+}
+```
+
+YAML Example:
+
+```yaml
+remote: false
+dockerizedBenchmark: true
+environment:
+  envoyPath: 'envoy'
+  outputDir: '/home/ubuntu/nighthawk_output'
+  testDir: '/home/ubuntu/nighthawk_tests'
+  v4only: true
+images:
+  reuseNhImages: true
+  nighthawkBenchmarkImage: 'envoyproxy/nighthawk-benchmark-dev:latest'
+  nighthawkBinaryImage: 'envoyproxy/nighthawk-dev:latest'
+  envoyImage: "envoyproxy/envoy-dev:f61b096f6a2dd3a9c74b9a9369a6ea398dbe1f0f"
+```
+
+In both examples, the envoy image being tested is a specific hash.  This hash can be replaced with "latest" to test the most recently created image against the previous image built from the prior Envoys master commit.
+
+
+## Building Salvo
+
+```bash
+bazel build //:salvo
```

Comment:
Yeah maybe this might be good to also add to `ci/do_ci.sh` to make sure that what we document also executes without error.